modules/lang-painless/src/internalClusterTest/java/org/elasticsearch/painless/action/CrossClusterPainlessExecuteIT.java

```diff
@@ -0,0 +1,197 @@
+import org.elasticsearch.core.Strings;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.index.query.MatchAllQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.painless.PainlessPlugin;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.script.FilterScript;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.script.ScriptContext;
+import org.elasticsearch.script.ScriptType;
+import org.elasticsearch.test.AbstractMultiClustersTestCase;
+import org.elasticsearch.test.InternalTestCluster;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentType;
+import org.hamcrest.Matchers;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+
+/**
+ * Tests the PainlessExecuteAction against a local cluster with one remote cluster configured.
+ * Execute action tests are run against both the local cluster and the remote cluster.
+ */
+public class CrossClusterPainlessExecuteIT extends AbstractMultiClustersTestCase {
+
+    private static final String REMOTE_CLUSTER = "cluster_a";
+    private static final String LOCAL_INDEX = "local_idx";
+    private static final String REMOTE_INDEX = "remote_idx";
+    private static final String KEYWORD_FIELD = "my_field";
+
+    @Override
+    protected Collection<String> remoteClusterAlias() {
+        return List.of(REMOTE_CLUSTER);
+    }
+
+    @Override
+    protected boolean reuseClusters() {
+        return false;
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins(String clusterAlias) {
+        List<Class<? extends Plugin>> plugs = Arrays.asList(PainlessPlugin.class);
+        return Stream.concat(super.nodePlugins(clusterAlias).stream(), plugs.stream()).collect(Collectors.toList());
+    }
+
+    public void testPainlessExecuteAgainstLocalCluster() throws Exception {
+        setupTwoClusters();
+
+        Script script = new Script(
+            ScriptType.INLINE,
+            Script.DEFAULT_SCRIPT_LANG,
+            Strings.format("doc['%s'].value.length() <= params.max_length", KEYWORD_FIELD),
+            Map.of("max_length", 4)
+        );
+        ScriptContext<?> context = FilterScript.CONTEXT;
+
+        PainlessExecuteAction.Request.ContextSetup contextSetup = createContextSetup(LOCAL_INDEX);
+        PainlessExecuteAction.Request request = new PainlessExecuteAction.Request(script, context.name, contextSetup);
+
+        ActionFuture<PainlessExecuteAction.Response> actionFuture = client(LOCAL_CLUSTER).admin()
+            .cluster()
+            .execute(PainlessExecuteAction.INSTANCE, request);
+
+        assertBusy(() -> assertTrue(actionFuture.isDone()));
```

Comment:
do we need this assertBusy? isn't the following actionGet enough?

Reply:
You are correct. I removed the assertBusy.